tasks.php

```diff
@@ -17,7 +17,8 @@
 
 $tasks_url = $code_url . "/" . basename(__FILE__);
 
-$requester_u_id = $userP['u_id'];
+$user = User::load_current();
```

Comment:
I don't think I see any issues, but here you introduce $user. If it was already in scope you could be overriding it. Would it be safer to do $requested_u_id = User::load_current()->u_id;
Again, probably not an issue.

Reply:
If `$user` was already in scope here we have problems as that shouldn't be a global variable -- we actually set that variable in various locations in the code assuming it will be used in a local context.